### PR TITLE
Crew update

### PR DESCRIPTION
**Pull request by satmandu**
- Fixes slow package checking during build
- Adds git_branch functionality

- The `gemacs` package is updated to work with this, but is currently uncompiled.  We should fix this, but do try testing this repo to see how it works for you before deployment.
```
crew update
export CREW_TESTING_BRANCH=git2
export CREW_TESTING_REPO=https://github.com/satmandu/chromebrew.git
CREW_TESTING=1 crew update
crew build gemacs
# to revert
crew update
```

Works properly:
- [x] x86_64
- [x] armv7l
- [x] i686

### CODE REVIEW

File: bin/crew [hunk 1]
```diff
@@ -757,22 +763,17 @@ def prepare_package (destdir)
...
+      sed 's,.filelist,,g' | sed 's,#{CREW_META_PATH},,g'"
+    conflicts << %x[#{@conflictscmd}].chomp.split(" ")
+    conflicts.reject!(&:empty?)
     unless conflicts.empty?
       puts "Unable to complete this build since there is a conflict with the same file in another package.".lightred
       pp = ''
       conflicts.each do |p, f|
         puts "\n#{p} package conflicts below:".lightred if p != pp
-        puts f.lightred
+        puts "#{f}".lightred
```
Comment (uberhacker) [inline bin/crew:776]: This line doesn't need to be changed.

File: packages/gemacs.rb [hunk 2]
```diff
@@ -38,56 +27,35 @@ class Gemacs < Package
...
 
   def self.install
-    Dir.chdir 'emacs_git' do
-      system 'make', "DESTDIR=#{CREW_DEST_DIR}", 'install'
-      system "install -Dm755 src/emacs #{CREW_DEST_PREFIX}/bin/gemacs"
-      system "install -Dm755 src/emacs-#{@_ver} #{CREW_DEST_PREFIX}/bin/gemacs-#{@_ver}"
-    end
+    system 'make', "DESTDIR=#{CREW_DEST_DIR}", 'install'
+    system "install -Dm755 src/emacs #{CREW_DEST_PREFIX}/bin/gemacs"
+    system "install -Dm755 src/emacs-#{@_ver} #{CREW_DEST_PREFIX}/bin/gemacs-#{@_ver}"
```
Comment (uberhacker) [inline packages/gemacs.rb:59]: Change to use `FileUtils`?

File: bin/crew (hunk 1 above)
Comment (satmandu) [reply bin/crew:779]: I know, but it was throwing errors unless I did that. :/

File: packages/gemacs.rb (hunk 2 above)
Comment (satmandu) [reply packages/gemacs.rb]: Done.

File: bin/crew (hunk 1 above)
Comment (satmandu) [reply bin/crew:779]: If it actually works for you feel free to change that back...

Comment (satmandu) [reply bin/crew]: works now

File: bin/crew [hunk 3]
```diff
@@ -757,16 +766,11 @@ def prepare_package (destdir)
...
-      packageName = File.basename packageList, '.filelist'
-      File.readlines(packageList).each do |line|
-        File.readlines('filelist').each do |item|
-          if line == item && packageName != @pkg.name
-            conflicts << [packageName, item.chomp]
-          end
-        end
-      end
-    end
+    @conflictscmd = "grep --exclude #{CREW_META_PATH}#{@pkg.name}.filelist \
```
Comment (uberhacker) [inline bin/crew:769]: Should we put a message here something like `puts "Checking for conflicts..."`?

Comment (satmandu) [reply bin/crew:770]: done. Does that look right?

Comment (uberhacker) [reply bin/crew:770]: That works!

**Pull request merged**